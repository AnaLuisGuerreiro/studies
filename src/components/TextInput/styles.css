.text-input {
  width: 100%;
  font-size: 25px;
  padding: 2px 10px;
}
