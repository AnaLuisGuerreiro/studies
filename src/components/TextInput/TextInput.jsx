import React from "react";

import "./styles.css";

export default function TextInput({ searchValue, handleChange }) {
  return (
    <input
      className="text-input"
      onChange={handleChange}
      value={searchValue}
      type="search"
      placeholder="Search for a post"
    />
  );
}
