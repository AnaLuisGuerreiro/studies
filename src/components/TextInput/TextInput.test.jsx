import { render, screen } from "@testing-library/react";

import TextInput from "./TextInput";
import userEvent from "@testing-library/user-event";

describe("<TextInput/>", () => {
  it("should have a value of searchValue", () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={"test"} />);

    const input = screen.getByPlaceholderText(/search for a post/i);
    expect(input.value).toBe("test");
  });

  it("should call handleChange function on each key pressed", () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/search for a post/i);

    const value = "o valor";

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it("should match snaptshot", () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} />);
    expect(container).toMatchSnapshot();
  });
});
