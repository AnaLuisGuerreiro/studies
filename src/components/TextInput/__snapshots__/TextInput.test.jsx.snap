// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should match snaptshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="Search for a post"
    type="search"
    value=""
  />
</div>
`;
